app/assets/javascripts/mumuki_laboratory/application/kids.js

```diff
@@ -219,9 +219,9 @@ mumuki.load(() => {
 
     _showOnSuccessPopup: function (data) {
       mumuki.kids._updateSubmissionResult(data.html);
-      mumuki.presenterCharacter.playAnimation('success_l', mumuki.kids._getCharacterImage());
+      mumuki.presenterCharacter.playAnimation(data.animation || 'success_l', mumuki.kids._getCharacterImage());
```

Comment:
As we are in a bit of a hurry it's ok, but we should keep the common logic in kids.js and create a primary.js for primary specific logic and avoid this "fallbacks" which are in fact regular behavior on non kindergarten layouts